test.js

```diff
@@ -26,6 +26,57 @@ test('get', t => {
 	t.true(dotProp.get({'foo\\.bar': true}, 'foo\\\\.bar'));
 	t.is(dotProp.get({foo: 1}, 'foo.bar'), undefined);
 
+	t.true(dotProp.get([true, false, false], '[0]'));
```

Comment:
The array access stuff should be in separate tests. This test is already way too large.